rfc/0075-faster-failover-and-configuration-push.md

```diff
@@ -0,0 +1,315 @@
+configuration updates. These notifications are lightweight and can be deduplicated by the server when the
+`DedupeNotMyVbucketClustermap` option is negotiated.
+
+#### Mixed Clusters
+
+In clusters where there is a mix of nodes with older server versions, meaning that some nodes do not acknowledge
+`ClustermapChangeNotificationBrief`, the respective connection should notify the configuration monitor about its lack of
+support for configuration pushes from the server. As a result, the monitor should utilize the old polling mechanism for
+this particular node instead.
+
+### Enhancements in Handling the `NotMyVbucket` Status
+
+Combination of `DedupeNotMyVbucketClustermap` and `ClustermapChangeNotificationBrief` allows to save traffic by not
+sending configuration, if SDK already seen the same revision, and also sends only pair of `Epoch`/`Revision`. So it is
+up to SDK to initiate configuration update once the non-empty payload returned along with `NotMyVbucket` status code.
+
+Several modifications are required in the SDK:
+1. The retry orchestrator should be able to retry an operation based on configuration updates rather than the timer signal.
+2. The configuration monitor should have the ability to throttle configuration requests due to the following reasons:
+   1. During rebalance, multiple operations may return a `NotMyVbucket` status, triggering a configuration refresh.
+   2. Since `ClustermapChangeNotificationBrief` will cause all connections to subscribe to updates and receive them, it is
+      necessary to account for potential high volumes of updates.
+
+Below is a diagram that illustrates an example of the SDK workflow, where the GET request is waiting for the arrival of
+a new configuration.
+
+```mermaid
+sequenceDiagram
+    autonumber
+    conn_1->>+kv_node_1: get("foo", vb=115)
+    kv_node_1->>-conn_1: NotMyVbucket(epoch=1, rev=11)
+    conn_1-->>+retry_orchestrator: pending(get, "foo", epoch=1, rev=11)
+    retry_orchestrator-->retry_orchestrator: put operation to wating queue
+    conn_1-->>+config_monitor: refresh configuration
+    config_monitor-->config_monitor: wait to throttle config requests
+    config_monitor->>+conn_2: get_config()
+    conn_2->>+kv_node_2: get_config()
+    kv_node_2->>-conn_2: configuration(epoch=1, rev=11)
+    conn_2->>-config_monitor: apply new configuration
+    config_monitor->>retry_orchestrator: purge waiting queue(epoch=1, rev=11)
+    retry_orchestrator->>conn_1: retry get("foo")
+    conn_1->>+kv_node_2: get("foo", vb=115)
+    kv_node_2->>-conn_1: Success()
+```
+
+# Language Specifics
+
+## Feature Checklist
+
+1. `GetClusterConfigWithKnownVersion` (`0x1d`). The SDK should always supply current configuration version if the
+   connection has acknowledged feature flag.
+
+2. `DedupeNotMyVbucketClustermap` (`0x1e`). The SDK should be ready that the KV engine will not repeat configuration
```

Comment:
I believe there is a related change to this where NVM's will only contain the revision number rather than the entire configuration.  Should we leverage that to simplify the overall SDK handling of errors?

Reply:
As it is said below, this flag affects all configuration updates, not only NMV. Could you clarify what do you mean by similification of the error handling?

Reply:
I've checked and indeed `NotMyVbucket` status either contains full config body (along with revision), or empty body meaning that the config has been seen already.

https://github.com/couchbase/kv_engine/blob/0a84f5b4f2377c9c6553e83091d0ab1b527cfd28/daemon/cookie.cc#L267-L300

I've updated RFC to send request to retry orchestrator in case of empty body